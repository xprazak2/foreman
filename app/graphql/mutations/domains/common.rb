module Mutations
  module Domains
    module Common
      extend ActiveSupport::Concern

      included do
        argument :name, String, required: false
        argument :fullname, String, required: false
        argument :dns_id, GraphQL::Types::ID, loads: Types::SmartProxy, required: false

        include ::Mutations::Concerns::TaxonomyArgs

        field :domain, Types::Domain, 'A domain.', null: true
      end
    end
  end
end
