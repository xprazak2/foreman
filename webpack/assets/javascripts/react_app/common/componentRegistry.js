import React from 'react';
import store from '../redux';
import PieChart from '../components/common/charts/PieChart/';
import StatisticsChartsList from '../components/statistics/StatisticsChartsList';
import PowerStatus from '../components/hosts/powerStatus/';
import NotificationContainer from '../components/notifications/';
import ToastsList from '../components/toastNotifications/';
import StorageContainer from '../components/hosts/storage/vmware/';
import _ from 'lodash';

let componentRegistry = () => {
    let registry = {
      PieChart: {
        type: PieChart,
        store: true,
        data: true
      },
      StatisticsChartsList: {
        type: StatisticsChartsList,
        store: true,
        data: true
      },
      PowerStatus: {
        type: PowerStatus,
        store: true,
        data: true
      },
      NotificationContainer: {
        type: NotificationContainer,
        store: true,
        data: true
      },
      ToastNotifications: {
        type: ToastsList,
        store: true,
        data: false
      }
    };

    return function () {
      let register = (name, component) => {
        if (registry[name]) {
          throw 'Component name already taken';
        }
        registry[name] = component;
        console.log('Component registered with name ' + name);
        return registry;
      };

      let getComponent = (name) => {
        return registry[name];
      };

      let registeredComponents = () => {
        return _.map(registry, (value, key) => {
          return key;
        }).join(', ');
      };

      let markup = (name, data, store) => {
        let currentComponent = getComponent(name);
        if (!currentComponent) {
          throw `Component not found:  ${name} among ${registeredComponents()}`
        }
        let ComponentName = currentComponent.type;
        return <ComponentName data={ currentComponent.data ? data : undefined } store={ currentComponent.store ? store : undefined } />;
      };

      return { register, getComponent, markup };
    }();
};

export default componentRegistry();
