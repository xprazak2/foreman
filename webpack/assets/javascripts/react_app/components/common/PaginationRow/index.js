import PaginationRow from './PaginationRow';

export default PaginationRow;
