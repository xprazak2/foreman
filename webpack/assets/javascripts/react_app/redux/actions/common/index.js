import API from '../../../API';

export const ajaxRequestAction = ({
  dispatch,
  requestAction,
  successAction,
  failedAction,
  url,
  item,
  headers = {},
  params = {},
}) => {
  dispatch({ type: requestAction, payload: item });
  return API.get(url, headers, params)
    .then(({ data }) => dispatch({ type: successAction, payload: { ...item, ...data } }))
    .catch(error => dispatch({ type: failedAction, payload: { error, item } }));
};
