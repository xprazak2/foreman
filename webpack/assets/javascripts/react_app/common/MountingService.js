import React from 'react';

// import StorageContainer from '../components/hosts/storage/vmware/';
// import StatisticsChartsList from '../components/statistics/StatisticsChartsList';
// import PowerStatus from '../components/hosts/powerStatus/';
// import NotificationContainer from '../components/notifications/';
// import ToastsList from '../components/toastNotifications/';
// import PieChart from '../components/common/charts/PieChart/';
import ReactDOM from 'react-dom';
import store from '../redux';
import componentRegistry from './componentRegistry';

export function mount(component, selector, data) {
  // const components = {
  //   PieChart: {
  //     type: PieChart,
  //     markup: <PieChart data={ data } />
  //   },
  //   StatisticsChartsList: {
  //     type: StatisticsChartsList,
  //     markup: <StatisticsChartsList store={store} data={data}/>
  //   },
  //   PowerStatus: {
  //     type: PowerStatus,
  //     markup: <PowerStatus store={store} data={data}/>
  //   },
  //   NotificationContainer: {
  //     type: NotificationContainer,
  //     markup: <NotificationContainer store={store} data={data} />
  //   },
  //   ToastNotifications: {
  //     type: ToastsList,
  //     markup: <ToastsList store={store} />
  //   },
  //   StorageContainer: {
  //     type: StorageContainer,
  //     markup: <StorageContainer store={store} data={data} />
  //   }
  // };
  const reactNode = document.querySelector(selector);

  if (reactNode) {
    ReactDOM.unmountComponentAtNode(reactNode);
    ReactDOM.render(componentRegistry.markup(component, data, store), reactNode);
  } else {
    const componentName = componentRegistry(component).type.name;

    // eslint-disable-next-line no-console
    console.log(`Cannot find \'${selector}\' element for mounting the \'${componentName}\'`);
  }
}
