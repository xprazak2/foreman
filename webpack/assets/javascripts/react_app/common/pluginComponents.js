import componentRegistry from './componentRegistry';
import _ from 'lodash';

export default function register(bundleNames) {
  console.log(bundleNames)
  _.forEach(bundleNames, (name) => {
    let pluginComponents = require(name + '/components');
    registerPluginComponents(pluginComponents);
  });
}

let registerPluginComponents = (pluginComponents) => {
  for (let [name, component] of Object.entries(pluginComponents)) {
    componentRegistry.register(name, component);
  }
};
