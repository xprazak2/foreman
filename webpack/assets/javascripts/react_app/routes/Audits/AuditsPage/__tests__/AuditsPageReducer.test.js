import {
  AUDITS_PAGE_SHOW_MESSAGE,
  AUDITS_PAGE_HIDE_MESSAGE,
  AUDITS_PAGE_CHANGE_PARAMS,
  AUDITS_PAGE_FETCH,
  AUDITS_PAGE_NEXT_PENDING,
  AUDITS_PAGE_NEXT_RESOLVED,
  AUDITS_PAGE_PREV_PENDING,
  AUDITS_PAGE_PREV_RESOLVED,
  AUDITS_PAGE_CLEAR_CACHE,
} from '../../constants';

import reducer from '../AuditsPageReducer';

import { testReducerSnapshotWithFixtures } from '../../../../common/testHelpers';
import { AuditsProps } from '../../../../components/AuditsList/__tests__/AuditsList.fixtures';

const fixtures = {
  'should return the initial state': {},
  'should handle AUDITS_PAGE_NEXT_PENDING': {
    action: {
      type: AUDITS_PAGE_NEXT_PENDING,
    },
  },
  'should handle AUDITS_PAGE_NEXT_RESOLVED': {
    action: {
      type: AUDITS_PAGE_NEXT_RESOLVED,
    },
  },
  'should handle AUDITS_PAGE_PREV_PENDING': {
    action: {
      type: AUDITS_PAGE_PREV_PENDING,
    },
  },
  'should handle AUDITS_PAGE_PREV_RESOLVED': {
    action: {
      type: AUDITS_PAGE_PREV_RESOLVED,
    },
  },
  'should handle AUDITS_PAGE_CLEAR_CACHE': {
    action: {
      type: AUDITS_PAGE_CLEAR_CACHE,
    },
  },
  'should handle AUDITS_PAGE_HIDE_MESSAGE': {
    action: {
      type: AUDITS_PAGE_HIDE_MESSAGE,
    },
  },
  'should handle AUDITS_PAGE_SHOW_MESSAGE': {
    action: {
      type: AUDITS_PAGE_SHOW_MESSAGE,
      payload: {
        showMessage: true,
        message: {
          text: __('No Audits found, please search again.'),
          type: 'empty',
        },
      },
    },
  },
  'should handle AUDITS_PAGE_FETCH': {
    action: {
      type: AUDITS_PAGE_FETCH,
      payload: {
        audits: AuditsProps.audits,
        itemCount: 21,
      },
    },
  },
  'should handle AUDITS_PAGE_CHANGE_PARAMS': {
    action: {
      type: AUDITS_PAGE_CHANGE_PARAMS,
      payload: {
        page: 2,
        perPage: 21,
        searchQuery: 'search',
      },
    },
  },
};

describe('AuditsPage reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
