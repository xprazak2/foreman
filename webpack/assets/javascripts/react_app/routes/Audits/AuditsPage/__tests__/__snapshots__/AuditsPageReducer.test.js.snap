// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditsPage reducer should handle AUDITS_PAGE_CHANGE_PARAMS 1`] = `
Object {
  "audits": Array [],
  "isFetchingNext": false,
  "isFetchingPrev": false,
  "itemCount": 0,
  "message": Object {},
  "nextAudits": Array [],
  "page": 2,
  "perPage": 21,
  "prevAudits": Array [],
  "searchQuery": "search",
  "showMessage": false,
}
`;

exports[`AuditsPage reducer should handle AUDITS_PAGE_CLEAR_CACHE 1`] = `
Object {
  "audits": Array [],
  "isFetchingNext": false,
  "isFetchingPrev": false,
  "itemCount": 0,
  "message": Object {},
  "nextAudits": Array [],
  "page": 1,
  "perPage": 20,
  "prevAudits": Array [],
  "searchQuery": "",
  "showMessage": false,
}
`;

exports[`AuditsPage reducer should handle AUDITS_PAGE_FETCH 1`] = `
Object {
  "audits": Array [
    Object {
      "action": "update",
      "action_display_name": "updated",
      "affected_locations": Array [
        Object {
          "name": "test_loc1",
          "url": "/locations/2-test_loc1/edit",
        },
        Object {
          "name": "test_loc2",
          "url": "/locations/6-test_loc2/edit",
        },
        Object {
          "name": "test_loc3",
          "url": "/locations/9-test_loc3/edit",
        },
      ],
      "affected_organizations": Array [
        Object {
          "name": "test_org1",
          "url": "/organizations/1-test_org1/edit",
        },
        Object {
          "name": "test_org2",
          "url": "/organizations/3-test_org2/edit",
        },
        Object {
          "name": "test_org3",
          "url": "/organizations/5-test_org3/edit",
        },
      ],
      "allowed_actions": Array [
        Object {
          "css_class": "btn btn-default",
          "title": "Host details",
          "url": "/hosts/host-foo.example.com",
        },
      ],
      "associated_id": null,
      "associated_name": null,
      "associated_type": null,
      "audit_title": "host-foo.example.com",
      "audit_title_url": "/audits?search=type+%3D+host+and+auditable_id+%3D+9",
      "auditable_id": 9,
      "auditable_name": "host-foo.example.com",
      "auditable_type": "Host::Base",
      "audited_changes": Object {
        "comment": Array [
          "",
          "This is info about host for audit",
        ],
        "root_pass": Array [
          "[redacted]",
          "[redacted]",
        ],
      },
      "audited_changes_with_id_to_label": Array [
        Object {
          "change": Array [
            Object {
              "css_class": "show-old",
              "id_to_label": "[redacted]",
            },
            Object {
              "css_class": "show-new",
              "id_to_label": "[redacted]",
            },
          ],
          "name": "Root pass",
        },
        Object {
          "change": Array [
            Object {
              "css_class": "show-old",
              "id_to_label": "[empty]",
            },
            Object {
              "css_class": "show-new",
              "id_to_label": "This is info about host for audit",
            },
          ],
          "name": "Comment",
        },
      ],
      "audited_type_name": "Host",
      "comment": null,
      "created_at": "2018-08-13 00:34:55 -1100",
      "id": 234,
      "remote_address": "127.0.0.1",
      "request_uuid": "c134239d-8ac3-494b-9962-35133fe153ba",
      "user_id": 4,
      "user_info": Object {
        "audit_path": "/audits?search=id+%3D+234",
        "display_name": "Admin ",
        "login": "admin",
        "search_path": "/audits?search=user+%3D+admin",
      },
      "user_type": null,
      "username": "Admin User",
      "version": 2,
    },
  ],
  "isFetchingNext": false,
  "isFetchingPrev": false,
  "itemCount": 21,
  "message": Object {},
  "nextAudits": Array [],
  "page": 1,
  "perPage": 20,
  "prevAudits": Array [],
  "searchQuery": "",
  "showMessage": false,
}
`;

exports[`AuditsPage reducer should handle AUDITS_PAGE_HIDE_MESSAGE 1`] = `
Object {
  "audits": Array [],
  "isFetchingNext": false,
  "isFetchingPrev": false,
  "itemCount": 0,
  "message": Object {},
  "nextAudits": Array [],
  "page": 1,
  "perPage": 20,
  "prevAudits": Array [],
  "searchQuery": "",
  "showMessage": false,
}
`;

exports[`AuditsPage reducer should handle AUDITS_PAGE_NEXT_PENDING 1`] = `
Object {
  "audits": Array [],
  "isFetchingNext": true,
  "isFetchingPrev": false,
  "itemCount": 0,
  "message": Object {},
  "nextAudits": Array [],
  "page": 1,
  "perPage": 20,
  "prevAudits": Array [],
  "searchQuery": "",
  "showMessage": false,
}
`;

exports[`AuditsPage reducer should handle AUDITS_PAGE_NEXT_RESOLVED 1`] = `
Object {
  "audits": Array [],
  "isFetchingNext": false,
  "isFetchingPrev": false,
  "itemCount": 0,
  "message": Object {},
  "nextAudits": Array [],
  "page": 1,
  "perPage": 20,
  "prevAudits": Array [],
  "searchQuery": "",
  "showMessage": false,
}
`;

exports[`AuditsPage reducer should handle AUDITS_PAGE_PREV_PENDING 1`] = `
Object {
  "audits": Array [],
  "isFetchingNext": false,
  "isFetchingPrev": true,
  "itemCount": 0,
  "message": Object {},
  "nextAudits": Array [],
  "page": 1,
  "perPage": 20,
  "prevAudits": Array [],
  "searchQuery": "",
  "showMessage": false,
}
`;

exports[`AuditsPage reducer should handle AUDITS_PAGE_PREV_RESOLVED 1`] = `
Object {
  "audits": Array [],
  "isFetchingNext": false,
  "isFetchingPrev": false,
  "itemCount": 0,
  "message": Object {},
  "nextAudits": Array [],
  "page": 1,
  "perPage": 20,
  "prevAudits": Array [],
  "searchQuery": "",
  "showMessage": false,
}
`;

exports[`AuditsPage reducer should handle AUDITS_PAGE_SHOW_MESSAGE 1`] = `
Object {
  "audits": Array [],
  "isFetchingNext": false,
  "isFetchingPrev": false,
  "itemCount": 0,
  "message": Object {
    "text": "No Audits found, please search again.",
    "type": "empty",
  },
  "nextAudits": Array [],
  "page": 1,
  "perPage": 20,
  "prevAudits": Array [],
  "searchQuery": "",
  "showMessage": true,
}
`;

exports[`AuditsPage reducer should return the initial state 1`] = `
Object {
  "audits": Array [],
  "isFetchingNext": false,
  "isFetchingPrev": false,
  "itemCount": 0,
  "message": Object {},
  "nextAudits": Array [],
  "page": 1,
  "perPage": 20,
  "prevAudits": Array [],
  "searchQuery": "",
  "showMessage": false,
}
`;
